### PR TITLE
Add support for preventing modal outside click close

### PR DESCRIPTION
**Pull request by Joemires**
This PR allows the control of modal close events on outside click
As discussed https://github.com/wireui/wireui/discussions/657, there's a use case to create a non-closable modal
I believe this would be useful

To create a non-closable modal, below describes how it can be done
``` html
<x-modal wire:model.defer="cardModal" align="center" prevent-outside-close="true">
    ...
</x-modal>
```

### CODE REVIEW

**Comment by ph7jack** [comment]
@Joemires Thanks for contributing, your idea looks good.
Please, add the attribute in the modal class.
`public bool $persistent = false;`

**Comment by Joemires** [comment]
Of course
Let me work that out

**Review by Joemires — COMMENTED** [review]
Persistence has been moved to modal class
Please review

File: resources/views/components/modal.blade.php [hunk 1]
```diff
@@ -24,7 +24,9 @@
             $blur => (bool) $blur
         ])
         x-show="show"
-        x-on:click="close"
+        @if(! $persistent)
+            x-on:click="close"
+        @endif
```
Comment (ph7jack) [inline resources/views/components/modal.blade.php]:
```suggestion
        @unless($persistent)
            x-on:click="close"
        @unless
```

File: resources/views/components/modal.blade.php [hunk 2]
```diff
@@ -35,7 +37,9 @@
 
     <div class="w-full min-h-full transform flex items-end justify-center mx-auto {{ $align }} {{ $maxWidth }}"
         x-show="show"
-        x-on:click.self="close"
+        @if(! $persistent)
+            x-on:click.self="close"
+        @endif
```
Comment (ph7jack) [inline resources/views/components/modal.blade.php]:
```suggestion
        @unless($persistent)
            x-on:click.self="close"
        @unless
```

**Review by ph7jack — DISMISSED** [review]
Looks good

**Comment by Joemires** [comment]
@PH7-Jack I noticed you changed the conditional check from if to unless
There's a little bug while closing it as you forgot to close it with endunless
I have opened a PR https://github.com/wireui/wireui/pull/761

**Comment by ph7jack** [comment]
@Joemires Very nice, thank you

**Pull request merged**